lib/reek/core/reference_collector.rb

```diff
@@ -12,14 +12,19 @@ def initialize(ast)
       end
 
       def num_refs_to_self
-        result = 0
-        [:self, :zsuper, :ivar, :ivasgn].each do |node_type|
-          @ast.each_node(node_type, STOP_NODES) { result += 1 }
-        end
-        @ast.each_node(:send, STOP_NODES) do |call|
-          result += 1 unless call.receiver
+        (explicit_self_calls + implicit_self_calls).size
+      end
+
+      private
+
+      def explicit_self_calls
+        [:self, :zsuper, :ivar, :ivasgn].flat_map do |node_type|
```

Comment:
It is important, otherwise this will return an `Array` of `Arrays` of calls (and won’t work); note that `each_node` returns an `Array`.